src/workerd/api/web-socket.c++

```diff
@@ -12,6 +12,30 @@
 
 namespace workerd::api {
 
+bool WebSocketProtocolError::decodeFromException(const kj::Exception& ex) {
+  kj::String magic = kj::heapString(magicFileValue);
+
+  auto maybeContext = ex.getContext();
+  while (true) {
+    KJ_IF_MAYBE(context, maybeContext) {
+      if (magic == context->file) {
+        code = context->line;
+        description = kj::heapString(context->description);
+        return true;
+      } else {
+        maybeContext = context->next;
+      }
+    } else {
+      return false;
+    }
+  }
+  return false;
```

Comment:
something like the following might be a bit simpler...

```suggestion
  while (maybeContext != nullptr) {
    auto& context = KJ_ASSERT_NONNULL(maybeContext);
    if (magic == context.file) {
      code = context.line;
      description = kj::str(context.description);
      return true;
    }
    maybeContext = context.next;
  }
  return false;
```

Reply:
Done.